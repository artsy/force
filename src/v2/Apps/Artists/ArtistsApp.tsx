import React from "react"

interface ArtistsAppProps {}

export const ArtistsApp: React.FC<ArtistsAppProps> = ({ children }) => {
  return <>{children}</>
}
