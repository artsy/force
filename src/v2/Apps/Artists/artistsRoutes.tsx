import loadable from "@loadable/component"
import { graphql } from "react-relay"
import { RouteConfig } from "found"

const ArtistsApp = loadable(() => import("./ArtistsApp"), {
  resolveComponent: component => component.ArtistsApp,
})

const ArtistsIndexRoute = loadable(() => import("./Routes/ArtistsIndex"), {
  resolveComponent: component => component.ArtistsIndexFragmentContainer,
})

const ArtistsByLetterRoute = loadable(
  () => import("./Routes/ArtistsByLetter"),
  {
    resolveComponent: component => component.ArtistsByLetterFragmentContainer,
  }
)

export const artistsRoutes: RouteConfig[] = [
  {
    path: "/artists2",
    getComponent: () => ArtistsApp,
    prepare: () => {
      return ArtistsApp.preload()
    },
    children: [
      {
        path: "",
        getComponent: () => ArtistsIndexRoute,
        prepare: () => {
          return ArtistsIndexRoute.preload()
        },
        query: graphql`
          query artistsRoutes_ArtistsQuery {
            featuredArtists: orderedSets(key: "homepage:featured-artists") {
              ...ArtistsIndex_featuredArtists
            }
            featuredGenes: orderedSets(key: "artists:featured-genes") {
              ...ArtistsIndex_featuredGenes
            }
          }
        `,
      },

      {
        path: "artists-starting-with-:letter([a-zA-Z])",
        getComponent: () => ArtistsByLetterRoute,
        prepare: () => {
          return ArtistsByLetterRoute.preload()
        },
        prepareVariables: (params, props) => {
          return {
            ...params,
            ...props,
            page: props.location.query.page ?? 1,
            size: 100,
          }
        },
        query: graphql`
          query artistsRoutes_ArtistsByLetterQuery(
            $letter: String!
            $page: Int
            $size: Int
          ) {
            viewer {
              ...ArtistsByLetter_viewer
                @arguments(letter: $letter, size: $size, page: $page)
            }
          }
        `,
      },
    ],
  },
]
