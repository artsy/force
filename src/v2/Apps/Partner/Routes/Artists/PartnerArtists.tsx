import { Box, Text, Button, Flex } from "@artsy/palette"
import React, { useEffect, useRef, useState } from "react"
import {
  createPaginationContainer,
  graphql,
  RelayPaginationProp,
} from "react-relay"
import { PartnerArtists_partner } from "v2/__generated__/PartnerArtists_partner.graphql"
import {
  PartnerArtistListFragmentContainer as PartnerArtistList,
  PartnerArtistListPlaceholder,
} from "../../Components/PartnerArtists"

const PAGE_SIZE = 20

export interface ArtistsRouteProps {
  partner: PartnerArtists_partner
  relay: RelayPaginationProp
}

function sleep(ms) {
  return new Promise(resolve => setTimeout(resolve, ms))
}

export const ArtistsRoute: React.FC<ArtistsRouteProps> = ({
  partner: { artistsConnection, distinguishRepresentedArtists, slug },
  relay,
}) => {
  const [artistsLoading, setArtistsLoading] = useState(relay.hasMore())
  const [isRefetching, setIsRefetching] = useState(false)
  const [artists, setArtists] = useState(artistsConnection)
  const errCounter = useRef(0)

  useEffect(() => {
    if (relay.hasMore()) {
      loadMoreArtists()
    }
  }, [])

  useEffect(() => {
    if ((!isRefetching || !artistsLoading) && artists !== artistsConnection) {
      setArtists(artistsConnection)
    }
  }, [artistsConnection, isRefetching, artistsLoading])

  const refetchArtists = () => {
    setArtistsLoading(true)
    setIsRefetching(true)
    errCounter.current = 0

    loadMoreArtists()
  }

  const loadMoreArtists = () => {
    relay.loadMore(PAGE_SIZE, async error => {
      if (error) {
        console.error(error)
        errCounter.current += 1
        // Wait before next try.
        await sleep(500)
      } else {
        errCounter.current = 0
      }

      if (errCounter.current >= 3) {
        setIsRefetching(false)
        setArtistsLoading(false)
        setArtists(artistsConnection)

        return
      }

      if (relay.hasMore()) {
        loadMoreArtists()
      } else {
        setIsRefetching(false)
        setArtistsLoading(false)
      }
    })
  }

  const isFirstLoading = !isRefetching && artistsLoading

  return (
    <Box mt={4}>
      <Text variant="title" mb={6}>
        Artists
      </Text>
      {!isFirstLoading && (
        <>
          <PartnerArtistList
            partnerSlug={slug}
            artists={artists.edges}
            distinguishRepresentedArtists={distinguishRepresentedArtists}
          />
          {(errCounter.current > 0 || isRefetching) && (
            <Flex flexDirection="column" mt={2} alignItems="center">
              <Button
                variant="secondaryOutline"
                loading={isRefetching}
                onClick={refetchArtists}
              >
                Load more artists
              </Button>
            </Flex>
          )}
        </>
      )}

      {isFirstLoading && <PartnerArtistListPlaceholder />}
    </Box>
  )
}

export const ARTISTS_QUERY = graphql`
  query PartnerArtistsQuery($partnerId: String!, $first: Int!, $after: String) {
    partner(id: $partnerId) {
      ...PartnerArtists_partner @arguments(first: $first, after: $after)
    }
  }
`

export const ArtistsPaginationContainer = createPaginationContainer(
  ArtistsRoute,
  {
    partner: graphql`
      fragment PartnerArtists_partner on Partner
        @argumentDefinitions(
          first: { type: "Int", defaultValue: 20 }
          after: { type: "String" }
        ) {
        slug
        distinguishRepresentedArtists
        artistsConnection(first: $first, after: $after)
          @connection(key: "PartnerArtistsQuery_artistsConnection") {
          edges {
            ...PartnerArtistList_artists
          }
        }
      }
    `,
  },
  {
    query: ARTISTS_QUERY,
    direction: "forward",
    getVariables(
      { partner: { slug: partnerId } },
      { cursor: after },
      { first }
    ) {
      return { after, first, partnerId }
    },
  }
)
