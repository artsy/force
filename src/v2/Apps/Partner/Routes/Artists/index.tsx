import React from "react"

export const ArtistsRoute: React.FC = () => {
  return <div>Artists</div>
}
