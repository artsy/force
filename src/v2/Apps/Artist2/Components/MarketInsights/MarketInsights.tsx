import { groupBy } from "lodash"
import React from "react"
import { createFragmentContainer, graphql } from "react-relay"
import styled from "styled-components"

import colors from "v2/Assets/Colors"
import { unica } from "v2/Assets/Fonts"
import TextLink from "v2/Components/TextLink"
import { Tooltip } from "v2/Components/Tooltip"

import { HelpIcon } from "@artsy/palette"

const MarketInsightsContainer = styled.div`
  ${unica("s14", "medium")};
`

const MarketInsightsDataContainer = styled.div`
  > div {
    display: inline-block;
    padding: 10px 30px;
  }
  border-bottom: 1px solid ${colors.grayRegular};
  border-top: 1px solid ${colors.grayRegular};
  padding: 20px;
  text-align: center;
`

const FeedbackContainer = styled.div`
  color: ${colors.graySemibold};
  padding: 8px 0;
  ${unica("s10")};

  a {
    ${unica("s10")};
  }
`

const TooltipContainer = styled.div`
  display: inline-block;
`

import { MarketInsights_artist } from "v2/__generated__/MarketInsights_artist.graphql"
interface Props extends React.HTMLProps<MarketInsights> {
  artist: MarketInsights_artist
}

const Categories = {
  "blue-chip": "Blue Chip",
  "top-emerging": "Top Emerging",
  "top-established": "Top Established",
}

const orderedCategories = ["blue-chip", "top-established", "top-emerging"]

const CategoryTooltipContent = {
  "blue-chip":
    "Blue chip galleries have multiple locations internationally and participate in major art fairs.",
  "top-emerging":
    "Top emerging dealers participate in curated, up-and-coming art fairs.",
  "top-established":
    "Top established galleries have been industry leaders in their region or specialty for decades.",
}

const groupedByCategories = edges => {
  return groupBy(edges, ({ node: partner }) => {
    let category
    Object.keys(Categories).forEach(key => {
      partner.categories.forEach(partnerCategory => {
        if (partnerCategory.slug === key) {
          category = key
        }
      })
    })
    return category
  })
}

export const highestCategory = edges => {
  const groups = groupedByCategories(edges)
  return orderedCategories.filter(
    category => groups[category] && groups[category].length > 0
  )[0]
}

export const hasSections = ({
  highlights,
  auctionResults,
  collections,
}: {
  highlights: { partnersConnection?: { edges: ReadonlyArray<any> } }
  auctionResults?: { edges: ReadonlyArray<any> }
  collections?: ReadonlyArray<any>
}) => {
  const { partnersConnection } = highlights

  // Is there a gallery representation section?
  if (
    partnersConnection &&
    partnersConnection.edges &&
    partnersConnection.edges.length > 0
  ) {
    return true
  }

  // Is there an auction highlights section?
  if (
    auctionResults &&
    auctionResults.edges &&
    auctionResults.edges.length > 0
  ) {
    return true
  }

  // Is there a permanent collections section?
  if (collections && collections.length > 0) {
    return true
  }

  return false
}

export class MarketInsights extends React.Component<Props, null> {
  renderGalleryCategory(categorySlug, partnerCount) {
    let introSentence
    const category = Categories[categorySlug]
    const categoryTooltipContent = CategoryTooltipContent[categorySlug]
    if (partnerCount > 1) {
      introSentence = "Represented by " + category.toLowerCase() + " galleries"
    } else {
      introSentence = "Represented by a " + category.toLowerCase() + " gallery"
    }

    return (
      <div>
        {introSentence}
        &nbsp;
        <TooltipContainer>
          {/* @ts-expect-error STRICT_NULL_CHECK */}
          <Tooltip message={categoryTooltipContent}>
            <span style={{ verticalAlign: "text-top" }}>
              <HelpIcon />
            </span>
          </Tooltip>
        </TooltipContainer>
      </div>
    )
  }

  // We group all partners that represent an artist by their relevant category, from the list above.
  // Assumption: these are mutually exclusive categories among a partner.
  renderGalleryRepresentation() {
    const { highlights } = this.props.artist
    // @ts-expect-error STRICT_NULL_CHECK
    const { partnersConnection } = highlights
    if (
      partnersConnection &&
      partnersConnection.edges &&
      partnersConnection.edges.length > 0
    ) {
      const highCategory = highestCategory(partnersConnection.edges)
      return (
        <div>
          {this.renderGalleryCategory(
            highCategory,
            groupedByCategories(partnersConnection.edges)[highCategory].length
          )}
        </div>
      )
    }
  }

  renderAuctionHighlight() {
    if (
      !this.props.artist.auctionResultsConnection ||
      // @ts-expect-error STRICT_NULL_CHECK
      this.props.artist.auctionResultsConnection.edges.length < 1
    ) {
      return null
    }
    // @ts-expect-error STRICT_NULL_CHECK
    const topAuctionResult = this.props.artist.auctionResultsConnection.edges[0]
      .node
    // @ts-expect-error STRICT_NULL_CHECK
    return <div>{topAuctionResult.price_realized.display} auction record</div>
  }

  renderPermanentCollection() {
    const { collections } = this.props.artist
    if (!collections || collections.length === 0) {
      return null
    }
    if (collections.length === 1) {
      return <div>Collected by a major museum</div>
    }
    return <div>Collected by major museums</div>
  }

  renderFeedbackLine() {
    return (
      <FeedbackContainer>
        Generated using partial data.&nbsp;
        <TextLink
          color={colors.graySemibold}
          // @ts-expect-error STRICT_NULL_CHECK
          underline
          href='mailto:productfeedback@artsy.net?subject=Feedback on "About the Artist" information'
        >
          Tell us what you think.
        </TextLink>
      </FeedbackContainer>
    )
  }

  render() {
    // @ts-expect-error STRICT_NULL_CHECK
    if (hasSections(this.props.artist)) {
      return (
        <MarketInsightsContainer>
          <MarketInsightsDataContainer>
            {this.renderAuctionHighlight()}
            {this.renderGalleryRepresentation()}
            {this.renderPermanentCollection()}
          </MarketInsightsDataContainer>
          {this.renderFeedbackLine()}
        </MarketInsightsContainer>
      )
    }
    return null
  }
}

export const MarketInsightsFragmentContainer = createFragmentContainer(
  // @ts-expect-error STRICT_NULL_CHECK
  MarketInsights,
  {
    artist: graphql`
      fragment MarketInsights_artist on Artist
        @argumentDefinitions(
          partnerCategory: {
            type: "[String]"
            defaultValue: ["blue-chip", "top-established", "top-emerging"]
          }
        ) {
        internalID
        collections
        highlights {
          partnersConnection(
            first: 10
            displayOnPartnerProfile: true
            representedBy: true
            partnerCategory: $partnerCategory
          ) {
            edges {
              node {
                categories {
                  slug
                }
              }
            }
          }
        }
        auctionResultsConnection(
          recordsTrusted: true
          first: 1
          sort: PRICE_AND_DATE_DESC
        ) {
          edges {
            node {
              price_realized: priceRealized {
                display(format: "0a")
              }
            }
          }
        }
      }
    `,
  }
)
