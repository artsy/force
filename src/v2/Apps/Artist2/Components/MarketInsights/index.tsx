import React from "react"
import { graphql } from "react-relay"

import { MarketInsightsContentsQuery } from "v2/__generated__/MarketInsightsContentsQuery.graphql"
import { SystemContextProps, withSystemContext } from "v2/Artsy"
import { SystemQueryRenderer as QueryRenderer } from "v2/Artsy/Relay/SystemQueryRenderer"
import { MarketInsightsFragmentContainer } from "./MarketInsights"

export interface Props extends SystemContextProps {
  artistID: string
}

class MarketInsightsContents extends React.Component<Props, null> {
  render() {
    const { artistID, relayEnvironment } = this.props
    return (
      <QueryRenderer<MarketInsightsContentsQuery>
        environment={relayEnvironment}
        query={graphql`
          query MarketInsightsContentsQuery($artistID: String!) {
            artist(id: $artistID) {
              ...MarketInsights_artist
            }
          }
        `}
        variables={{ artistID }}
        render={({ props }) => {
          if (props) {
            return <MarketInsightsFragmentContainer artist={props.artist} />
          } else {
            return null
          }
        }}
      />
    )
  }
}

export const Contents = withSystemContext(MarketInsightsContents)
