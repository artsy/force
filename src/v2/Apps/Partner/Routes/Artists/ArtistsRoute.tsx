import React, { useEffect } from "react"
import { Box, Text, themeProps } from "@artsy/palette"
import { Match } from "found"
import {
  PartnerArtistDetailsListRenderer,
  PartnerArtistDetailsRenderer,
  PartnerArtistsFragmentContainer,
} from "../../Components/PartnerArtists"
import { ArtistsRoute_partner } from "v2/__generated__/ArtistsRoute_partner.graphql"
import { PARTNER_NAV_BAR_HEIGHT } from "../../Components/NavigationTabs"
import { graphql } from "lib/graphql"
import { createFragmentContainer } from "react-relay"
import { Media } from "v2/Utils/Responsive"
import { usePartnerArtistsLoadingContext } from "../../Utils/PartnerArtistsLoadingContext"
import { scrollIntoView } from "v2/Utils/scrollHelpers"
import { useMatchMedia } from "v2/Utils/Hooks/useMatchMedia"
import { useNavBarHeight } from "v2/Components/NavBar/useNavBarHeight"

export interface ArtistsRouteProps {
  partner: ArtistsRoute_partner
  match: Match
}

export const ArtistsRoute: React.FC<ArtistsRouteProps> = ({
  partner,
  match,
}) => {
  const { mobile, desktop } = useNavBarHeight()
  const isMobile = useMatchMedia(themeProps.mediaQueries.xs)
  const { isLoaded } = usePartnerArtistsLoadingContext()

  useEffect(() => {
    if (match.params.artistId && isLoaded && isMobile !== null) {
      scrollIntoArtistDetails()
    }
  }, [isLoaded, isMobile])

  const scrollIntoArtistDetails = () => {
    const offset = PARTNER_NAV_BAR_HEIGHT + (isMobile ? mobile : desktop) + 20

    scrollIntoView({
      offset: offset,
      selector: "#jump--PartnerArtistDetails",
    })
  }

  return (
    <Box mt={4}>
      <Text variant="title" mb={6}>
        Artists
      </Text>
      <Media greaterThan="xs">
        <PartnerArtistsFragmentContainer
          scrollTo={{
            selector: "#jump--PartnerArtistDetails",
            offset: PARTNER_NAV_BAR_HEIGHT + desktop + 20,
          }}
          partner={partner}
        />
      </Media>
      <Media at="xs">
        <PartnerArtistsFragmentContainer
          scrollTo={{
            selector: "#jump--PartnerArtistDetails",
            offset: PARTNER_NAV_BAR_HEIGHT + mobile + 20,
          }}
          partner={partner}
        />
      </Media>

      {match.params.artistId ? (
        <PartnerArtistDetailsRenderer
          partnerId={match.params.partnerId}
          artistId={match.params.artistId}
        />
      ) : (
        <PartnerArtistDetailsListRenderer partnerId={match.params.partnerId} />
      )}
    </Box>
  )
}

export const ArtistsRouteFragmentContainer = createFragmentContainer(
  ArtistsRoute,
  {
    partner: graphql`
      fragment ArtistsRoute_partner on Partner {
        ...PartnerArtists_partner
      }
    `,
  }
)
