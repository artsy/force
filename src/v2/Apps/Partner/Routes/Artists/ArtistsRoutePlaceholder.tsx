import React from "react"
import { Box, SkeletonText } from "@artsy/palette"
import { PartnerArtistListPlaceholder } from "../../Components/PartnerArtists"
import { PartnerArtistDetailsPlaceholder } from "../../Components/PartnerArtists/PartnerArtistDetailsPlaceholder"

export const ArtistsRoutePlaceholder = () => {
  return (
    <Box mt={4}>
      <SkeletonText variant="title" mb={6}>
        Artists
      </SkeletonText>
      <PartnerArtistListPlaceholder />
      <PartnerArtistDetailsPlaceholder />
    </Box>
  )
}
