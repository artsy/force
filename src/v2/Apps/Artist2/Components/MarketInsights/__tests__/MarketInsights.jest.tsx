import { mount } from "enzyme"
import "jest-styled-components"
import React from "react"
import renderer from "react-test-renderer"
import { MarketInsightsFragmentContainer } from "../MarketInsights"
import { MarketInsightsArtists } from "./Fixtures/Artists.fixture"

describe("MarketInsights", () => {
  describe("snapshots", () => {
    it("renders correctly", () => {
      const marketInsights = renderer
        .create(
          <MarketInsightsFragmentContainer
            artist={MarketInsightsArtists[0] as any}
          />
        )
        .toJSON()
      expect(marketInsights).toMatchSnapshot()
    })
  })

  describe("unit", () => {
    it("renders market data if present", () => {
      const component = mount(
        <MarketInsightsFragmentContainer
          artist={MarketInsightsArtists[0] as any}
        />
      )
      expect(component.text()).toMatch("$63m auction record")
      expect(component.text()).toMatch("Represented by blue chip galleries")
      expect(component.text()).toMatch("Collected by major museums")
      expect(component.text()).toMatch(
        "Generated using partial data.Â Tell us what you think."
      )
    })

    it("renders nothing if no market data", () => {
      const component = mount(
        <MarketInsightsFragmentContainer
          artist={MarketInsightsArtists[1] as any}
        />
      )
      expect(component.html()).toBe(null)
    })

    describe("#renderGalleryCategory", () => {
      it("prints single results", () => {
        const component = mount(
          <MarketInsightsFragmentContainer
            artist={MarketInsightsArtists[0] as any}
          />
        )
        const {
          props: { children },
        } = (component.instance() as any).renderGalleryCategory("blue-chip", 1)

        expect(children[0]).toMatch("Represented by a blue chip gallery")
      })

      it("prints plural results", () => {
        const component = mount(
          <MarketInsightsFragmentContainer
            artist={MarketInsightsArtists[0] as any}
          />
        )
        const {
          props: { children },
        } = (component.instance() as any).renderGalleryCategory(
          "top-emerging",
          2
        )

        expect(children[0]).toMatch("Represented by top emerging galleries")
      })
    })
  })
})
