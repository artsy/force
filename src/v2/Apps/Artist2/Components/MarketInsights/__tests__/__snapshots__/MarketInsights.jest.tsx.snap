// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarketInsights snapshots renders correctly 1`] = `
.c6 {
  font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
  font-size: 15px;
  line-height: 1.25em;
  -webkit-font-smoothing: antialiased;
  color: #666666;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c3 {
  font-family: Unica77LLWebRegular,Arial,serif;
  -webkit-font-smoothing: antialiased;
  font-size: 12px;
  line-height: 1.4em;
  display: inline-block;
  position: relative;
  cursor: help;
  margin: 0 0.5em;
  width: 14px;
  height: 14px;
  margin-bottom: -2px;
}

.c3::before {
  display: block;
  top: 0;
  left: 0;
  position: absolute;
  content: "";
  z-index: 2;
  width: 14px;
  height: 14px;
  border-radius: 50%;
  line-height: 15px;
  text-align: center;
  color: white;
}

.c3:hover::before {
  visibility: hidden;
}

.c3:hover::after {
  opacity: 1;
  z-index: 3;
  visibility: visible;
}

.c3::after {
  display: block;
  top: 0;
  bottom: inherit;
  left: 0;
  right: inherit;
  position: absolute;
  visibility: hidden;
  text-align: left;
  z-index: 1;
  margin: -10px 0 0 -10px;
  width: 300px;
  color: #666666;
  font-weight: normal;
  background-color: white;
  padding: 20px;
  opacity: 0;
  margin: 0;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
  border: 1px solid #e5e5e5;
  content: "Blue chip galleries have multiple locations internationally and participate in major art fairs.";
}

.c4 {
  position: relative;
}

.c0 {
  font-family: Unica77LLWebMedium,Arial,serif;
  -webkit-font-smoothing: antialiased;
  font-size: 14px;
  line-height: 1.4em;
}

.c1 {
  border-bottom: 1px solid #e5e5e5;
  border-top: 1px solid #e5e5e5;
  padding: 20px;
  text-align: center;
}

.c1 > div {
  display: inline-block;
  padding: 10px 30px;
}

.c5 {
  color: #666666;
  padding: 8px 0;
  font-family: Unica77LLWebRegular,Arial,serif;
  -webkit-font-smoothing: antialiased;
  font-size: 10px;
  line-height: 1.4em;
}

.c5 a {
  font-family: Unica77LLWebRegular,Arial,serif;
  -webkit-font-smoothing: antialiased;
  font-size: 10px;
  line-height: 1.4em;
}

.c2 {
  display: inline-block;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div>
      $63m
       auction record
    </div>
    <div>
      <div>
        Represented by blue chip galleries
         
        <div
          className="c2"
        >
          <div
            className="c3"
          >
            <span
              style={
                Object {
                  "verticalAlign": "text-top",
                }
              }
            >
              <svg
                className="c4"
                fill="black100"
                height="18px"
                viewBox="0 0 18 18"
                width="18px"
              >
                <title>
                  More info
                </title>
                <path
                  d="M9 1.889A7.111 7.111 0 1 1 9 16.11 7.111 7.111 0 0 1 9 1.89zM9 1a8 8 0 1 0 0 16A8 8 0 0 0 9 1zm-.658 9.92v-.142c0-1.111.214-1.458 1.307-2.418.405-.294.66-.753.693-1.253a1.182 1.182 0 0 0-1.298-1.254 1.422 1.422 0 0 0-1.448 1.6H6.573a2.391 2.391 0 0 1 2.534-2.4 2.089 2.089 0 0 1 2.32 2c0 1.716-2.143 2.01-2.143 3.69v.177h-.942zm-.089 2.027V11.8H9.41v1.147H8.253z"
                  fillRule="nonzero"
                />
              </svg>
            </span>
          </div>
        </div>
      </div>
    </div>
    <div>
      Collected by major museums
    </div>
  </div>
  <div
    className="c5"
  >
    Generated using partial data. 
    <a
      className="c6"
      href="mailto:productfeedback@artsy.net?subject=Feedback on \\"About the Artist\\" information"
      target="_self"
    >
      Tell us what you think.
    </a>
  </div>
</div>
`;
